week-4/Homework/mandatory/4-groceries.js

```diff
@@ -1,12 +1,95 @@
-// You're going shopping, and you need a shopping list.
-// 1. Update your groceryList with the items you need: Potatoes, Orange Juice and Rice.
-// 2. Loop through the groceryList object to gather the item properties into the groceriesToBuy array.
-// 3. Then use console.log() to print out the list. It should print ['Potatoes', 'Orange Juice', 'Rice']
+/*
+As you you can have an Array of Objects, you can also store Arrays in Objects.
+In this exercise, you'll practice:
+ - How to loop through the properties (keys) of an Object and read its values.
+ - How to access Array stored inside Object.
+ - How to access to a specific property of an array and set it.
 
-let groceriesToBuy = [];
+You're going shopping, and you need a shopping list. You've already created your weekly meal plan
+that contains the missing ingredients to your menus. It is stored in the "weeklyMealPlan" object.
 
-let groceryList = {
-  item1: "",
-  item2: "",
-  item3: ""
+Complete the exercises below.
+*/
+
+// Here is your 
+let weeklyMealPlan = {
+  monday: ["Cheese", "Eggs", "Tomato", "Paprika", "Leek"],
+  tuesday: ["Wrap", "Tuna", "Canned beans", "Cheese", "Carrot", "Aubergine"],
+  wednesday: ["Orange Juice", "Apple", "Ananas", "Black tea"],
+  thursday: ["Lamb", "Salt", "Bulgur", "Potato"],
+  fridray: ["Rice milk", "Blueberries", "Porridge", "Banana", "Cinnamon"],
+  saturday: ["Olive oil", "Potato", "Salmon", "Asparagus"],
+  sunday: []
+};
+
+/*
+Exercise 1:
+  Loop through the weekly meal plan object to gather weakly ingredients into the weeklyGroceriesToBuy array.
+  Then use console.log() to print out the list.
+*/
+
+console.log('Ex-1 Weekly groceries items to buy!')
+let weeklyGroceriesToBuy = [];
+for (dayOfWeek in weeklyMealPlan) {
```

Comment:
```suggestion
for (const dayOfWeek in weeklyMealPlan) {
```
Ensure you're using a keyword when you declare a variable like `dayOfWeek` in your `for` loop